### ISSUE TITLE
RTSP handshake error 60 after driver update

### ISSUE THREAD

**Issue by cgutman** [moonlight-stream/nvidia-gamestream-issues#32]
A few users have reported RTSP handshake error 60 after updating their drivers. The issue is not affecting the majority of Moonlight users (including myself), but it affects some users very consistently. It must be something specific to their client, host, or streaming settings that is triggering the issue reliably.

Error 60 is `ETIMEDOUT` which would indicate either a connection timeout or a timeout waiting for a reply from the host. In some cases, GeForce Experience will fail to reply to RTSP messages while waiting for NvStreamer.exe to start. Specific invalid streaming settings (such as resolutions below 256x256) are known to cause NvStreamer.exe to crash, so that may be what's happening here.

Affected users, please provide the following info:
- What was the last driver version to work and the first driver version not to work?
- Do you have "hardware accelerated GPU scheduling" enabled on your host PC?
- Attach your logs from `C:\ProgramData\NVIDIA Corporation\NvStream` after trying to stream and getting the RTSP error

**Comment by anoncoder2022** [comment]
I am currently experiencing this issue but only over online attempts, not in local network. I am able to connect successfully when at home in my network, but attempts to connect from anywhere else causes me to get this error. I have ensured the ports are open and available through my router / modem.

- I am not sure what driver it worked last on, but currently it is happening on Nvidia Geforce 512.15 
- hardware accelerated GPU scheduling is enabled on host PC
- [NvStreamerOld.log](https://github.com/moonlight-stream/nvidia-gamestream-issues/files/8834178/NvStreamerOld.log)

**Comment by cgutman** [comment]
Ah, if it happens on remote streaming but not local streaming that definitely points to some issue with port forwarding. I suspect the issue is with UDP port 48000. GeForce Experience will stall the RTSP handshake until it gets a response on that port, so that can lead to timeouts if it's not forwarded properly.

Can you confirm that you have that UDP 48000 forwarded properly?

**Comment by mikesells** [comment]
I have the same issue as above. My belief is that no forwarding is necessary if I'm using something like ZeroTier correct? I have run moonlight tester and it says all is well.

**Comment by cgutman** [comment]
That's correct

**Comment by jvmesa** [comment]
Same error for me. For more info:
Only online. LAN / WLAN works fine.
Client = Ipad Pro 11 IpadOS 15.5 M
Driver = 516.59
Moonlight Internet Streaming Tester pass correct (attached).
Launching the game/app I see client connects with PC and initiate the process correctly,The game/app is launched in the PC but at the same time the client stops returning this error code. The game/app remains working on the PC so I should close it locally.

Thanks

![TestOK](https://user-images.githubusercontent.com/108762975/177406027-f3544ac6-7bdc-4ee9-a6f2-4a3c717d8109.jpg)

**Comment by dmiyamasu** [comment]
I am experiencing this issue too. Any updates?

**Comment by billyjacoby** [comment]
I'm having this same issue using a Wireguard VPN that works for a number of other services.

**Comment by aur3l14no** [comment]
Same, I'm having this same issue using a tailscale.

**Comment by cyturralde** [comment]
Does anyone have a fix for this?

**Comment by gtttt33399** [comment]
Same issue: I access my LAN via OpenVPN on my NAS from outside my network and receive the rtsp 60 error when trying to connect to Sunshine on my PC. Also happens when I connect via TailScale directly to the PC. Moonlight client reports no issues when I select the "test network" option and all other remote services are working via the VPN. I've got the port forwarding set up properly on my router, etc. running a 4090 on the 531.79 driver. 

If I click on an app via Moonlight while on my VPN, the program starts up (which I can see when I hop on my PC later) but then the 60 error hits.

This used to work fine with GFE in the past (although I couldn't tell you what driver version, unfortunately).

Attaching the log here as well — everything until "CLIENT DISCONNECTED" is me connecting via my laptop on the home network, everything afterwards until [2023:05:21:22:31:59] is me trying to connect from a mobile network and OpenVPN.

[sunshinelog.txt](https://github.com/moonlight-stream/nvidia-gamestream-issues/files/11526460/sunshinelog.txt)

EDIT: Think I managed to sort it out by editing my OpenVPN config. Enabled redirect gateway (which is fine for my purposes — might not be for others!) and manually defined the DNS servers.

**Comment by thelazurus** [comment]
Running into the same issue through Tailscale.

**Comment by joshdee** [comment]
Same exact issue as well. But I have a router that is capable of self-hosting a VPN through my whole home Wi-Fi, specifically Amplifi Alien. It's a minor issue but I *really* want to use Tailscale. ^^

**Comment by SgtSweetieBelle** [comment]
same issue. router port forwarded and everything :(

**Comment by ClassyCar** [comment]
I've got the same issue. Has anyone found a solution yet? I'm using Zerotier one and trying to connect to my PC from my iPhone.

**Comment by adnanjamil-frontend** [comment]
Same issue here, any solutions? Connecting with iPhone using Zerotier

**Comment by TristanSamson** [comment]
Anyone found a solution? Using Tailscale to connect from my Iphone. (Sidenote: it seems like a lot of us having problems comes from trying to connect from Iphone. I'll try with my laptop later to see if that's the case)

**Comment by boyphongsakorn** [comment]
any update? i have a same issue with Tailscale (Mac and Android)

**Comment by slrgt** [comment]
Still having the same issue. Happens on tailscale and zerotier but not if im local. Im on an iphone

**Comment by alanwake47** [comment]
Same issue. Iphone 14 Pro Max and Ipad Pro M4 11in. Seems like and iOS specific thing. 
Works flawlessly on android. I even used my iPhone as a hotspot for my Pixel 7 and was able to connect.
I use Tailscale, & Sunshine to remotely connect

**Comment by knittycatkitty** [comment]
Same issue, iPhone 14 Pro Max on iOS 18

can verify it works on Galaxy A51 on Android separately though, must be an iOS issue

**Comment by newlandk** [comment]
iPhone 16 Pro on iOS 18.1 Public Beta 2

I'm hoping this may guide towards a more permanent solution assuming this is able to solve anyone else's issue.

TLDR: Providing Wireguard a 0's route (0.0.0.0/0) in the Allowed IPs list fixed the "error 60" issue and is working as expected now.

### Test scenarios
| Network Connection | Wireguard Allowed IPs | Result | 
|-|-|-|
|VPN via Guest Wireless Network| 192.168.0.0/24 (Same network Sunshine host) | Working|
|VPN over 5G | 192.168.0.0/24 | NOT Working (error 60)|
|VPN over 5G Hotspot | 192.168.0.0/24 | NOT Working (error 60)|
|VPN over 5G | 192.168.0.0/24, 0.0.0.0/0 | Working  |
|VPN over 5G Hotspot | 192.168.0.0/24, 0.0.0.0/0 | Working |

In all of these scenarios I can confirm with packet captures that traffic is traversing the tunnel between my phone and the host running Sunshine. In the scenarios where it's not working I can see that it's receiving 200 OK from the server and bidirectional traffic to TCP ports 47989 and 47984, but never establishes the UDP stream.

My peer endpoint is listed as an IPv6 address when on cellular and most likely traversing a 6to4 gateway. I'm thinking this may be a catalyst as the first test traversing my guest network over VPN is IPv4 only and worked as intended with only the internal networks routed over the VPN. Plumbing in the 0's route effectively making the VPN a full tunnel seems to be changing the behavior allowing it to establish the stream.

**Comment by sopresig** [comment]
I can successfully stream within the local area network, but when I connect to my sunshine through an FRP tunnel, I have confirmed that the port forwarding for all seven ports is configured correctly, and I can also ping through. However, I encounter ERR 60.

**Comment by DailyDisco** [comment]
Any solution to this yet?

Does UDnP have to be enabled?

**Comment by knittycatkitty** [comment]
I haven’t been able to attend to this issue as I’m currently missing a
charger for my Lenovo laptop and want to source a new/used one locally.

I’ll update when I get my hands on one.

Justin R.

**Comment by ThatJRadGuy** [comment]
Same issue here trying to connect my iphone 15 over 5g to sunshine through tailscale. It was working a few days ago, and since i updated ios and tailscale (previous version was from March this year) on my pc i’m getting this error

**Comment by LookToWinward** [comment]
Running into this same issue

**Comment by RageMC** [comment]
> iPhone 16 Pro on iOS 18.1 Public Beta 2
> 
> I'm hoping this may guide towards a more permanent solution assuming this is able to solve anyone else's issue.
> 
> TLDR: Providing Wireguard a 0's route (0.0.0.0/0) in the Allowed IPs list fixed the "error 60" issue and is working as expected now.
> ### Test scenarios
> Network Connection 	Wireguard Allowed IPs 	Result
> VPN via Guest Wireless Network 	192.168.0.0/24 (Same network Sunshine host) 	Working
> VPN over 5G 	192.168.0.0/24 	NOT Working (error 60)
> VPN over 5G Hotspot 	192.168.0.0/24 	NOT Working (error 60)
> VPN over 5G 	192.168.0.0/24, 0.0.0.0/0 	Working
> VPN over 5G Hotspot 	192.168.0.0/24, 0.0.0.0/0 	Working
> 
> In all of these scenarios I can confirm with packet captures that traffic is traversing the tunnel between my phone and the host running Sunshine. In the scenarios where it's not working I can see that it's receiving 200 OK from the server and bidirectional traffic to TCP ports 47989 and 47984, but never establishes the UDP stream.
> 
> My peer endpoint is listed as an IPv6 address when on cellular and most likely traversing a 6to4 gateway. I'm thinking this may be a catalyst as the first test traversing my guest network over VPN is IPv4 only and worked as intended with only the internal networks routed over the VPN. Plumbing in the 0's route effectively making the VPN a full tunnel seems to be changing the behavior allowing it to establish the stream.

How did you set it up and how do i do it through tailscale or zeriotier or just directly through my router

**Comment by EzioAuditore2020** [comment]
same issue as of today, my Samsung S24+ can stream just fine but my iPad Pro over phone hotspot (5G) won't connect. Must be an iOS thing. still no solution after 3 years? WTF?

**Comment by RageMC** [comment]
> same issue as of today, my Samsung S24+ can stream just fine but my iPad Pro over phone hotspot (5G) won't connect. Must be an iOS thing. still no solution after 3 years? WTF?

I solved the issue with my IPad and iPhone. Just go to Tailscale on your pc if you have one and turn on exit node in the desktop app and then make sure to turn it on in the website and when you go to your iPhone or iPad app on IOS there will be a highlight above saying exit node enable or disable. Enable it and you should be able to use Moonlight with your data. I don’t know why this is possible but it just works. This also allowed me to use Razer’s remote play app too.

**Comment by ThatJRadGuy** [comment]
> > same issue as of today, my Samsung S24+ can stream just fine but my iPad Pro over phone hotspot (5G) won't connect. Must be an iOS thing. still no solution after 3 years? WTF?
> 
> I solved the issue with my IPad and iPhone. Just go to Tailscale on your pc if you have one and turn on exit node in the desktop app and then make sure to turn it on in the website and when you go to your iPhone or iPad app on IOS there will be a highlight above saying exit node enable or disable. Enable it and you should be able to use Moonlight with your data. I don’t know why this is possible but it just works. This also allowed me to use Razer’s remote play app too.
> 

Interesting. I read somewhere else about the exit node trick but I couldn't get it to work. Maybe I missed one of the steps you mentioned. I'll give it another go next time i'm on the PC and report back.

**Comment by ThatJRadGuy** [comment]
After following the steps above to enable the exit node, I've managed to make it through the RTSP handshake without an error on 5G. I think I missed a step or two the last time I tried it.
Thanks! @RageMC

**Comment by RageMC** [comment]
> After following the steps above to enable the exit node, I've managed to make it through the RTSP handshake without an error on 5G. I think I missed a step or two the last time I tried it. Thanks! [@RageMC](https://github.com/RageMC)

You are welcome. I was just clicking through random things on the desktop app and website and I accidentally did it not knowing that would actually fix it. I wish there was a way to let more people know of this because I know not a lot of people are going to see the message I gave and not much people are gonna do a lot of research on it. Even the guy who said to setup a 0s route didn’t really elaborate and I tried doing it with ZeroTier and Tailscale but I couldn’t figure out how. This is so much easier.

**Comment by itz-coffee** [comment]
I am having the same issue without a VPN, seems to be inconsistent and the only solution is to keep retrying to connect

Moonlight 6.1.0
Sunshine v0.23.1
MacOS Sequoia 15.3.2

**Comment by pjackiewicz111-bit** [comment]
i haved the same problem idk what you guys using bu i wanted to connect form phone to pc by sunshine and moonlight and i just reinstaled sunshine on pc and reseted my phone and pc and now its working.
